src/sema/solana_accounts.rs

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use phf::phf_set;
+
+/// These are the accounts that we can collect from a contract and that Anchor will populate
+/// automatically if their names match the source code description:
+/// https://github.com/coral-xyz/anchor/blob/06c42327d4241e5f79c35bc5588ec0a6ad2fedeb/ts/packages/anchor/src/program/accounts-resolver.ts#L54-L60
+pub static CLOCK_ACCOUNT: &str = "clock";
```

Comment:
Why are those needed if they all are in the HashSet below?

Reply:
I think this section could have a better design, but I could think of any simple way to do this. I need these strings in many parts of the code to add the proper accounts to the IDL. Rewriting the same string repeatedly is prone to errors.

At the same time, these are reserved keywords for account names, so developers cannot declare them as account names.

I needed a way to get the string and check if a string is one reserved word.

Reply:
Sounds like these should be enums variants then? You can implement (or even better derive)  from and to static methods